docs/notebooks/qubit_mapping_guide.ipynb

```diff
@@ -0,0 +1,796 @@
+    }
+   },
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Qubit Mapping\n",
+    "\n",
+    "\n",
+    "![image.png](attachment:image.png)\n",
+    "\n",
+    "Mapping is the process of taking qubits from an algorithmic \"abstract\" framework to the physical ones existing on an actual quantum backend. There are two considerations when mapping a quantum circuit to a backend: the fact that most physical backends have limited qubit connectivity and the quality of the physical qubits. Selection of *good* qubits together with minimizing the number of SWAP gates arising from limited qubit connectivity are often orthogonal requirements.\n",
+    "\n",
+    "Therefore, we separate qubit mapping into **two** distinct sequential processes (which also reflects that they are solved by two separate classical algorithms):\n",
+    "\n",
+    "- **Routing**: Finding the least number of `SWAP` gates needed to route quantum information given limited connectivity.\n",
+    "  \n",
+    "- **Layout**: Finding the best qubits (given some metric or cost function) that respects the topology given by the Routing pass.\n",
+    "\n",
+    "The second process is the primary topic of discussion in this notebook.\n",
+    "\n",
+    "Note that both of these problems are NP-hard, and hence we use stochastic algorithms SABRE and VF2, respectively. For better performance, especially on larger number of qubits, consider running each several times, using the metrics\n",
```

Comment:
"heuristic algorithms" might be a better word to describe these algorithms instead of stochastic algorithms.